### DISCUSSION TITLE
Requesting Guidance on Understanding Source Code Details Related to the 'task' Option

### DISCUSSION THREAD

**Discussion by RedHeartSecretMan** [openai/whisper#1140] [Q&A] [answered]
### issue
**I'm currently working on a project and want to add some additional features, such as expanding the translation scope. The current version can translate from X to English, but I want to expand it to include translations to Chinese or other languages. However, when I started reading the source code for further learning, I found some code that is difficult to understand. Can you give me some advice as a developer?**

> For instance, in the 655 lines of code for decoding, there is a conditional statement that checks whether self.options.task is equal to "lang_id". It may seem unusual to use "lang_id" instead of "translate" or "transcribe"

**thanks!**

**Comment by jongwook** [comment, marked as answer]
The model is not trained for that; it occasionally does it when you set `task="translate"` and `language` as the target language, but it's mostly due to imperfect language id in the training data, and it won't be very reliable.

`"lang_id"` was intended for the case just doing the language identification and no transcription, but it isn't very well tested or documented. It's not relevant for what you're trying to do, though.
